lib/std/math/gdc.zig

```diff
@@ -0,0 +1,26 @@
+/// Find the greatest common divisor (GCD) of two integers.
+const std = @import("std");
+const expect = std.testing.expect;
+
+// Greatest common divisor
+fn gcd(a: anytype, b: anytype) @TypeOf(a) {
+    if (a < 0 or b < 0) @compileError("gcd is only defined for positive numbers (integer)");
+
+    var x = a;
```

Comment:
You need to annotate the types of `x`, `y` and `m` as `@TypeOf(a, b)`. Consider the call `gcd(@as(u8, 1) @as(u16, 300))`. This will currently cause safety-checked UB on the first loop iteration, since it attempts to assign `300` to `x` which has type `u8` (so cannot store that value). Using `@TypeOf(a, b)` as explicit types for all these values fixes that.